app/controllers/lists_controller.rb

```diff
@@ -62,7 +62,7 @@ def destroy
   private
     # Use callbacks to share common setup or constraints between actions.
     def set_list
-      @list = List.find_by(slug: params[:id])
+      @list = List.find_by(slug: params[:list_slug] || params[:id])
```

Comment:
yea I agree.   The routing syntax is confusing to me and I wasn't sure how to get destroy to pass :list_slug instead of params[:id].  Either way though, do you want to stick with :list_slug as the identifier?